src/backend/rating_app/views/analytics.py

```
@@ -37,7 +37,7 @@ class AnalyticsViewSet(viewsets.ViewSet):
         parameters=COURSES_LIST_QUERY_PARAMS,
         responses=R_ANALYTICS,
     )
-    def list(self, request, *args, **kwargs):
+    def list(self, request: Request, *args, **kwargs):
```

Ruff (0.14.4)

Undefined name `Request`

(F821)